supertokens_python/recipe/session/interfaces.py

```diff
@@ -589,20 +599,23 @@ def __init__(self, is_valid: bool, reason: Optional[Dict[str, Any]] = None):
 
 
 class SessionClaimValidator(ABC):
-    def __init__(self, id_: str):
+    def __init__(
+        self,
+        id_: str,
+        claim: SessionClaim[_T],  # pyright: ignore[reportInvalidTypeVarUse]
+    ) -> None:
         self.id = id_
-        self.claim: Optional[
-            SessionClaim[Any]
-        ] = None  # Child class must set this if required.
+        self.claim = claim
 
     @abstractmethod
     async def validate(
         self, payload: JSONObject, user_context: Dict[str, Any]
     ) -> ClaimValidationResult:
         pass
 
-    def should_refetch(  # pylint: disable=no-self-use
+    @abstractmethod
+    def should_refetch(
         self, payload: JSONObject, user_context: Dict[str, Any]
     ) -> MaybeAwaitable[bool]:
-        _, __ = payload, user_context
-        return False
+        # TODO: Confirm that MaybeAwaitable actually makes the function async
+        pass
```

Comment:
Make this non abstract, and by default, make it throw an error. The child claim class should override this if required, else this should not be called.

Reply:
Done.